File: TelerikUIExamples/TelerikUIExamples/BindWithCustomObject.m
//
//  BindWithCustomObject.m
//  TelerikUIExamples
//
//  Copyright (c) 2013 Telerik. All rights reserved.
//

#import "BindWithCustomObject.h"
#import <TelerikUI/TelerikUI.h>

@interface CustomObject : NSObject

@property (nonatomic, assign) NSInteger objectID;
@property (nonatomic, assign) CGFloat value1;
@property (nonatomic, assign) CGFloat value2;
@property (nonatomic, assign) CGFloat value3;

@end

@implementation CustomObject
@end

@implementation BindWithCustomObject
{
    TKChart *_chart;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    _chart = [[TKChart alloc] initWithFrame:self.exampleBoundsWithInset];
    _chart.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    [self.view addSubview:_chart];
    
    NSMutableArray *data = [[NSMutableArray alloc] init];
    for (int i = 0; i<=5; i++) {
        CustomObject *obj = [[CustomObject alloc] init];
        obj.objectID = i;
        obj.value1 = arc4random() % 100;
        obj.value2 = arc4random() % 100;
        obj.value3 = arc4random() % 100;
        [data addObject:obj];
    }
    
    [_chart addSeries:[[TKChartAreaSeries alloc] initWithItems:data forKeys:@{ @"dataXValue": @"objectID", @"dataYValue": @"value1"}]];
    [_chart addSeries:[[TKChartAreaSeries alloc] initWithItems:data forKeys:@{ @"dataXValue": @"objectID", @"dataYValue": @"value2" }]];
    [_chart addSeries:[[TKChartAreaSeries alloc] initWithItems:data xValueKey:@"objectID" yValueKey:@"value3"]];
    
    TKChartStackInfo *stackInfo = [[TKChartStackInfo alloc] initWithID:@(1) withStackMode:TKChartStackModeStack];
    for (int i =0; i<_chart.series.count; i++) {
        TKChartSeries *series = _chart.series[i];
        series.selectionMode = TKChartSeriesSelectionModeSeries;
        series.stackInfo = stackInfo;
    }

    [_chart reloadData];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/BindWithCustomObject.swift
//
//  BindWithCustomObject.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation

class CustomObject : NSObject
{
    var objectID:Int?
    var value1:Float?
    var value2:Float?
    var value3:Float?

    override func valueForKey(key: String) -> AnyObject? {
        switch key {
            case "objectID":
                return self.objectID
            case "value1":
                return self.value1
            case "value2":
                return self.value2
            case "value3":
                return self.value3
        default:
            break
        }
       return super.valueForKey(key)
    }
}

class BindWithCustomObject: ExampleViewController {
    
    let chart = TKChart()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        chart.frame = self.exampleBoundsWithInset
        chart.autoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight
        self.view.addSubview(chart)
        
        var data = [CustomObject]()
        for i in 0..<6 {
            let obj = CustomObject()
            obj.objectID = i
            obj.value1 = Float(arc4random() % (100))
            obj.value2 = Float(arc4random() % (100))
            obj.value3 = Float(arc4random() % (100))
            data.append(obj)
        }
        
        chart.addSeries(TKChartAreaSeries(items: data, forKeys: ["dataXValue": "objectID", "dataYValue": "value1"]))
        chart.addSeries(TKChartAreaSeries(items: data, forKeys: ["dataXValue": "objectID", "dataYValue": "value2"]))
        chart.addSeries(TKChartAreaSeries(items: data, xValueKey: "objectID", yValueKey: "value3"))
        
        var stackInfo = TKChartStackInfo(ID: 1, withStackMode: TKChartStackMode.Stack)
        for i in 0..<chart.series().count {
            let series = chart.series()[i] as! TKChartSeries
            series.selectionMode = TKChartSeriesSelectionMode.Series
            series.stackInfo = stackInfo
        }

        chart.reloadData()
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}